include/qdmi/device.h

```diff
@@ -593,6 +638,249 @@ void QDMI_device_job_free(QDMI_Device_Job job);
+ * querying environment sensors, i.e. temperature or power.
+ *
+ *  The typical workflow for a device environment sensor query is as follows:
+ *  - Create an environment sensor query with @ref
+ * QDMI_device_session_create_environmentsensor_query.
+ *  - Set parameters for the environment sensor query with @ref
+ * QDMI_device_environmentsensor_query_set_parameter
+ *  - Submit the environment sensor query to the device with @ref
+ * QDMI_device_environmentsensor_query_submit.
+ *  - Check the status of the environment sensor query with @ref
+ * QDMI_device_environmentsensor_query_check_status.
+ *  - Wait for the environment sensor query to finish with @ref
+ * QDMI_device_environmentsensor_query_wait.
+ *  - Retrieve the results of the environment sensor query with @ref
+ * QDMI_environmentsensor_query_get_results.
+ *  - Free the environment sensor query with @ref
+ * QDMI_device_environmentsensor_query_free when it is no longer used.
+ *
+ *  @{
+ */
+
+/**
+ * @brief A handle for a device environment sensor query.
+ * @details An opaque pointer to a type defined by the device that encapsulates
+ * all information about an environment sensor query on a device.
+ * @remark Implementations of the underlying type will want to store the session
+ * handle used to create the environment sensor query in the environment sensor
+ * query handle to be able to access the session information when needed.
+ * @see QDMI_EnvironmentSensor_Query for the client-side job handle.
+ */
+typedef struct QDMI_Device_EnvironmentSensor_Query_impl_d
+    *QDMI_Device_EnvironmentSensor_Query;
+/**
+ * @brief Create an environment sensor query.
+ * @details This is the main entry point for a driver to create an environment
+ * sensor query for a device. The returned handle can be used throughout the
+ * @ref device_environmentsensor_query_interface
+ * "device environment sensor query interface" to refer to the environment
+ * query.
+ * @param[in] session The session to create the environment sensor query on.
+ * Must not be @c NULL.
+ * @param[out] query A pointer to a handle that will store the created
+ * environment sensor query. Must not be @c NULL. The environment sensor query
+ * must be freed by calling
+ * @ref QDMI_device_environmentsensor_query_free when it is no longer used.
+ * @return @ref QDMI_SUCCESS if the environment sensor query was successfully
+ * created.
+ * @return @ref QDMI_ERROR_INVALIDARGUMENT if @p session or @p query are @c
+ * NULL.
+ * @return @ref QDMI_ERROR_BADSTATE if the session is not in a state allowing
+ * the creation of an environment sensor query, for example, because the session
+ * is not initialized.
+ * @return @ref QDMI_ERROR_PERMISSIONDENIED if the device does not allow using
+ * the @ref device_environmentsensor_query_interface "device environment sensor
+ * query interface" for the current session.
+ * @return @ref QDMI_ERROR_FATAL if environment sensor query creation failed due
```

Comment:
```suggestion
 * @return @ref QDMI_ERROR_FATAL if the environment sensor query creation failed due
```
I have the slight feeling that there is missing a `the`.